### PR TITLE
Update valid documentation for net_ping

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### SUMMARY

Update valid documentation for net_ping module 
Fixes: https://github.com/ansible-collections/ansible.netcommon/issues/100

Depends-On: ansible-collections/vyos.vyos#140
Depends-On: ansible-collections/arista.eos#183
Depends-On: ansible-collections/junipernetworks.junos#161

##### ISSUE TYPE

- Docs Pull Request
- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: docs/ansible.netcommon.net_ping_module.rst [hunk 1]
```diff
@@ -18,8 +18,7 @@ Version added: 1.0.0
 Synopsis
 --------
 - Tests reachability using ping from network device to a remote destination.
-- For Windows targets, use the :ref:`ansible.windows.win_ping <ansible.windows.win_ping_module>` module instead.
-- For targets running Python, use the :ref:`ansible.builtin.ping <ansible.builtin.ping_module>` module instead.
```
Comment (NilashishC) [inline docs/ansible.netcommon.net_ping_module.rst:22]: @ashwini-mhatre I'm not sure if I totally understand why these lines are being removed and replaced with `wait_for`. Do you mind giving a quick overview of this?

Comment (ashwini-mhatre) [reply docs/ansible.netcommon.net_ping_module.rst:22]: So this is fix for https://github.com/ansible-collections/ansible.netcommon/issues/100

Comment (ashwini-mhatre) [reply docs/ansible.netcommon.net_ping_module.rst:22]: The behavior of built-in ping and net-ping is different. For net-ping we can use wait_for to achieve same behavior.

Comment (NilashishC) [reply docs/ansible.netcommon.net_ping_module.rst:22]: `net_ping` (or any of the network platform specific ping modules) is used to check connectivity from the target node to another node and not between controller and target node. From my understanding, `wait_for` is not equivalent to net_ping and instead is a substitute for achieving ICMP ping check like behaviour. Neither network ping modules, not builtin ping module actually uses ICMP pings. Please correct me if I misunderstood something.

Comment (ashwini-mhatre) [reply docs/ansible.netcommon.net_ping_module.rst:22]: builtin ping is not using ICMP pings please check https://docs.ansible.com/ansible/latest/collections/ansible/builtin/ping_module.html

Comment (ashwini-mhatre) [reply docs/ansible.netcommon.net_ping_module.rst:22]: also please check documentation for https://docs.ansible.com/ansible/latest/collections/ansible/netcommon/net_ping_module.html#ansible-collections-ansible-netcommon-net-ping-module if you check return values you will get to know this uses ICMP ping

Comment (NilashishC) [reply docs/ansible.netcommon.net_ping_module.rst:22]: > builtin ping is not using ICMP pings please check https://docs.ansible.com/ansible/latest/collections/ansible/builtin/ping_module.html

Yes, that is what I meant too.



> also please check documentation for https://docs.ansible.com/ansible/latest/collections/ansible/netcommon/net_ping_module.html#ansible-collections-ansible-netcommon-net-ping-module if you check return values you will get to know this uses ICMP ping

It does use ICMP pings, but not between controller and target. Instead, it tests the reachability between target and another node.

Comment (ashwini-mhatre) [reply docs/ansible.netcommon.net_ping_module.rst:22]: ack

**Review by NilashishC — APPROVED** [review]

**Review by samccann — APPROVED** [review]
LGTM

**Comment by ashwini-mhatre** [comment]
recheck

**Review by Qalthos — APPROVED** [review]

**Comment by ashwini-mhatre** [comment]
recheck

**Comment by Qalthos** [comment]
recheck

**Review by rohitthakur2590 — APPROVED** [review]

**Pull request merged**